Inventory properties to implementation

Faithcaio opened the pull request:
**SpongeAPI** | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/1935)

 - [x] SlotPos
 - [x] SlotSide
 - [x] StringProperty
 - [x] EquipmentSlotType
 - [x] GuiIdProperty
 - [x] Identifiable
 - [x] InventoryCapacity
 - [x] SlotIndex
 - [x] InventoryDimension
 - [x] InventoryTitle

Removed because unused / covered by other property
 - [x] ArmorSlotType
 - [x] MappedIntProperty
 - [x] AcceptsItems

Cybermaxke commented:
Are you also going to move the `Property`s that aren't related to inventories? The builder can then be moved to the basic `Property` class.
https://github.com/SpongePowered/SpongeAPI/tree/bleeding/src/main/java/org/spongepowered/api/data/property

Faithcaio commented:
I'd rather not in this PR.
I actually never used the other Properties.
I'm only breaking the stuff I know.

Cybermaxke commented:
Hmm, ok. Can you move the `InventoryProperty.Builder` to `Property.Builder` instead, so we don't have to break the API twice when moving over all the other properties.

Cybermaxke commented on src/main/java/org/spongepowered/api/item/inventory/InventoryProperty.java:
```diff
@@ -25,13 +25,42 @@
+     * @param <B>
+     */
+    interface Builder<V, T extends InventoryProperty<?, V>, B extends Builder<V, T, B>> extends ResettableBuilder<T, B> {
+
```
A builder method for the key?

Faithcaio replied:
I'm considering removing that.
It is always the default key atm.

Cybermaxke replied:
Yeah, maybe remove the key from the API, haven't seen it used anywhere that it isn't default. If we remove this then it should be done for all the `Property`s.
@gabizou thoughts?

gabizou commented on src/main/java/org/spongepowered/api/entity/vehicle/minecart/ContainerMinecart.java:
```diff
@@ -31,7 +31,7 @@
  * are {@link ChestMinecart} and {@link HopperMinecart}.
  */
-public interface ContainerMinecart<M extends ContainerMinecart<M>> extends Minecart, Carrier, CarriedInventory<M> {
+public interface ContainerMinecart<M extends ContainerMinecart<M>> extends Minecart, Carrier {
```
`M` doesn't look to be used.

Faithcaio committed:
Move inventory properties to implementation
ContainerMinecart is not an inventory but its Carrier
Inventory is now a PropertyHolder

Cybermaxke commented on src/main/java/org/spongepowered/api/data/Property.java:
```diff
@@ -222,4 +225,45 @@ public static Operator defaultOperator() {
+     * @param <T> The inventory property type
+     * @param <B> The builder type
+     */
+    interface Builder<V, T extends InventoryProperty<?, V>, B extends Builder<V, T, B>> extends ResettableBuilder<T, B> {
```
Generics need to be changed to target `Property` instead of `InventoryProperty`.
